invenio_app_rdm/upgrade_scripts/migrate_13_0_to_14_0.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# Invenio-App-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record migration script from InvenioRDM 13.0 to 14.0.
+
+Disclaimer: This script is intended to be executed *only once*, namely when
+upgrading from InvenioRDM 13.0 to 14.0!
+If this script is executed at any other time, probably the best case scenario
+is that nothing happens!
+
+
+This script has been tested with following data:
+
+- record
+    - internal_notes
+"""
+
+from click import secho
+from invenio_access.permissions import system_identity
+from invenio_rdm_records.proxies import current_rdm_records_service as records_service
+from invenio_search.engine import dsl
+
+
+def run_upgrade(has, migrate_record, migrate_draft):
+    """Run upgrade."""
+    # Handle published records
+    published_records = records_service.search(
+        system_identity,
+        params={"allversions": True, "include_deleted": True},
+        extra_filter=has,
+    )
+    for result in published_records.hits:
+        try:
+            migrate_record(result)
+        except Exception as error:
+            secho(f"> Error {repr(error)}", fg="red")
+            error = f"Record {result['id']} failed to update"
+
+    # Handle draft records
+    draft_records = records_service.search_drafts(
+        system_identity,
+        params={"allversions": True},
+        extra_filter=has,
+    )
+    for result in draft_records.hits:
+        try:
+            migrate_draft(result)
+        except Exception as error:
+            secho(f"> Error {repr(error)}", fg="red")
+            error = f"Draft {result['id']} failed to update"
+
+
+def run_update_for_resource_type():
+    """Run update for resource type."""
+
+    def migrate_resource_type_in_record(hit_result):
+        """
+        Update resource type from event to publication-dissertation.
+        We go through the service layer to automatically trigger the DOI update and re-indexing.
+        """
+        print(f"Updating resource type for {hit_result['id']}")
+        record = records_service.read_latest(system_identity, hit_result["id"])
+        if record.data["metadata"]["resource_type"]["id"] != "event":
+            secho(
+                f"Skipping record <{record.id}> because it doesn't have resource-type 'event'!",
+                fg="yellow",
+            )
+            return
+
+        draft = records_service.edit(system_identity, record.id)
+        if draft.data["metadata"] != record.data["metadata"]:
```

Comment:
I am not very sure that this comparison will work fine. We should instead implement the logic that:
1. if published, but no draft, then edit -> publish
2. if published, with a draft, can we update and save both, without triggering a version conflict? (to be fully tested)
3. if only draft, update

Reply:
> I have added the line `draft._record.fork_revision_id = record._record.revision_id`, and tested it in my local instance, the changes are propagating correctly!

Did you manage to test that you can successfully publish the draft after the change to the revisions?

Reply:
Yes, it works.